spark/src/test/scala/org/apache/spark/sql/delta/DeltaDataFrameWriterV2Suite.scala

```diff
@@ -676,9 +675,10 @@ class DeltaDataFrameWriterV2Suite
       spark.sql(s"CREATE TABLE $table(id bigint, p int) USING delta PARTITIONED BY (p)")
       def verifyNotImplicitCasting(f: => Unit): Unit = {
         val e = intercept[DeltaAnalysisException](f)
-        assert(e.getErrorClass == "DELTA_FAILED_TO_MERGE_FIELDS")
-        assert(Utils.exceptionString(e)
-          .contains("Failed to merge incompatible data types LongType and IntegerType"))
+        checkError(
+          exception = e.getCause.asInstanceOf[DeltaAnalysisException],
+          errorClass = "DELTA_MERGE_INCOMPATIBLE_DATATYPE",
+          parameters = Map("currentDataType" -> "LongType", "updateDataType" -> "IntegerType"))
```

Comment:
We typically return the SQL type names. Is there a reason not to do this here? FYI: @MaxGekk

Reply:
I see. Changing the message format to use `toSQLType` string, will take additional effort to fix all references, beyond the scope of this work. Thanks.